python/ray/train/torch.py

```diff
@@ -425,6 +526,29 @@ def accelerate() -> None:
         )
 
 
+@PublicAPI(stability="beta")
+def prepare_optimizer(optimizer: torch.optim.Optimizer) -> torch.optim.Optimizer:
+    """Wraps optimizer to support automatic mixed precision.
```

Comment:
Can we mention/point to `accelerate` here? Since technically this only works if the user has called `accelerate(amp=True)`

Reply:
I believe this should work even without `accelerate(amp=True)`, and if not, is it possible have it work even in that case?

Reply:
`prepare_optimizer` should work without `accelerate(amp=True)`.

Reply:
Maybe I'm missing something, but doesn't the user need to explicitly set `amp=True` to enable automatic mixed precision (since the default value is `False`)?

Reply:
Ah I see what you're saying.

But I'm not sure if adding it to the docstring of `prepare_optimizer` would be the right place to mention amp. `prepare_optimizer` is agnostic to any accelerations, right? Ultimately the user guide should be the place to specify this.